src/main/java/me/partlysanestudios/partlysaneskies/garden/VisitorLogbookStats.kt

```diff
@@ -0,0 +1,158 @@
+            val lineOffersAccepted = noPlcwList.find{ it.contains("Offers Accepted: ") } ?: break
+            seenStats[rarityIndex] += lineTimesVisited.split(" ").last().replace(",", "").replace(".", "").toInt()
+            acceptedStats[rarityIndex] += lineOffersAccepted.split(" ").last().replace(",", "").replace(".", "").toInt()
+            seenStats[0] += lineTimesVisited.split(" ").last().replace(",", "").replace(".", "").toInt()
+            acceptedStats[0] += lineOffersAccepted.split(" ").last().replace(",", "").replace(".", "").toInt()
+        }
+        getString(seenStats, acceptedStats)
+    }
+
+    private fun getString(seenStats: MutableList<Int>, acceptedStats: MutableList<Int>) {
+        for (indexInt in 0..(tiers.size - 1)) { //"'RaNgEtO' oR ThE '..' cALL sHoULd bE RePLaCeD wiTh 'UnTiL'" SHUT THE FUCK UP INTELLIJ LET ME CODE HOWEVER THE FUCK I WANT TO
+            theBaseString += "\n${tiers[indexInt]}: [${seenStats[indexInt]} | ${acceptedStats[indexInt]} | ${Math.abs(seenStats[indexInt] - acceptedStats[indexInt])}]"
+        }
+    }
+
+    private fun isVisitorLogbook(): Boolean {
+        val gui = PartlySaneSkies.minecraft.currentScreen ?: return false
+        if (gui !is GuiChest) {
+            return false
+        }
+
+        val inventories = PartlySaneSkies.getSeparateUpperLowerInventories(gui)
+        val logbook = inventories[0]
+
+        return StringUtils.removeColorCodes(logbook.getDisplayName().getFormattedText()).contains("Visitor's Logbook")
+    }
+
+    val window = Window(ElementaVersion.V2)
+
+    val box = UIRoundedRectangle(5f)
+            .setColor(Color(0, 0, 0, 0))
+            .setChildOf(window)
+    
+    val image = ThemeManager.getCurrentBackgroundUIImage()
+            .setChildOf(box)
+    
+    val pad = 5
+    var textComponent: UIWrappedText = UIWrappedText() childOf box
+
+    @SubscribeEvent
+    fun renderInformation(event: GuiScreenEvent.BackgroundDrawnEvent) {
+        if (!isVisitorLogbook()) {
+            box.hide()
+            return
+        }
+        if (!PartlySaneSkies.config.visitorLogbookStats) {
+            return
+        }
+
+        box.unhide(true)
+        box.setX(widthScaledConstraint(700f))
+            .setY(CenterConstraint())
+            .setWidth(widthScaledConstraint(250f))
+            .setHeight(widthScaledConstraint(300f))
+
+        image.setX(CenterConstraint())
+            .setY(CenterConstraint())
+            .setWidth(PixelConstraint(box.getWidth()))
+            .setHeight(PixelConstraint(box.getHeight()))
+        
+        textComponent.setX(widthScaledConstraint(pad.toFloat()))
+            .setTextScale(widthScaledConstraint(1f))
+            .setY(widthScaledConstraint(2 * (pad.toFloat())))
+            .setWidth(PixelConstraint(box.getWidth() - (2 * (pad.toFloat()))))
+
+
+        var textString = "§2Garden Visitor Stats (for current page):\n§7Format: [visit | accept | deny + pending]\n"
```

Comment:
I feel like its better with all past tense buts that up to you

Reply:
past tense = text is going to go out of the box

Reply:
Lmao thats one way to solve that